src/data_warehouse/import/import.ts

```diff
@@ -0,0 +1,104 @@
+import {BaseDomainClass, NakedDomainClass} from "../../common_classes/base_domain_class";
+import {IsDate, IsDefined, IsNumber, IsOptional, IsString, IsUUID} from "class-validator";
+import {Type} from "class-transformer";
+
+/*
+    Import represents an import record in the Deep Lynx database and the various
+    validations required for said record to be considered valid.
+ */
+export default class Import extends BaseDomainClass {
+    @IsOptional()
+    @IsUUID()
+    id?: string
+
+    @IsUUID()
+    data_source_id?: string
+
+    @IsOptional()
+    @IsString()
+    status_message?: string
+
+    @IsString()
+    status: "ready" | "processing" | "error" | "stopped" | "completed" = "ready"
```

Comment:
Reason for not using @IsIn?

Reply:
In this case the data type is an actual ENUM on the backend. Since we rely heavily on the actual type name, and not so much as a "supported" value type thing, I decided to keep this as it was originally.